### ISSUE TITLE
Question about the variables

### ISSUE THREAD

**Issue by RogerRafa** [graspnet/AnyDexGrasp#4]
Thanks for your great work. However, I am confused about the variables in MinkowskiGraspNetMultifingerType1. Could you please explain the meaning and the difference of the `num_multifinger_depth` and the `num_two_finger_depth`?

**Comment by longshot-pting** [comment]
> Thanks for your great work. However, I am confused about the variables in MinkowskiGraspNetMultifingerType1. Could you please explain the meaning and the difference of the `num_multifinger_depth` and the `num_two_finger_depth`?

hey, have you get right allegro pose when running this code? I get a (1024,15) shape allegro pose and my code here. Thank you and hope for replying!
`def get_allegro_pose_from_raw_pointcloud(cfgs, pointcloud, extrinsic=np.eye(4), voxel_size=0.005):
    """
    输入：
    - cfgs: 包含模型路径等参数
    - pointcloud: numpy.ndarray of shape (N, 3)，单位为米
    - extrinsic: 相机外参（将点云变换到世界坐标），默认为单位矩阵
    - voxel_size: 用于稀疏体素化的大小

    输出：
    - dict: 包含 Allegro 抓取姿态信息（rotation, translation, score, type, etc.）
    """

    assert pointcloud.shape[1] == 3, "Point cloud must be of shape (N, 3)"

    # 1. 点云变换到世界坐标
    points = transform_point_cloud(pointcloud.astype(np.float32), extrinsic)

    # 2. 构造稀疏体素张量
    coords = np.ascontiguousarray(points / voxel_size, dtype=int)
    _, idxs = ME.utils.sparse_quantize(coords, return_index=True)
    coords = coords[idxs]
    points = points[idxs]
    coords_batch, points_batch = ME.utils.sparse_collate([coords], [torch.from_numpy(points)])

    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
    # sinput = ME.SparseTensor(features=points_batch, coordinates=coords_batch, device=device)
    sinput = ME.SparseTensor(
        features=points_batch.float(),  # 强制转换为 float32
        coordinates=coords_batch.int(),  # 通常是 int32，这里保险起见显式写出
        device=device
    )

    # 3. 模型推理
    net = get_net(cfgs.checkpoint_path, use_v2=cfgs.use_graspnet_v2)
    allegro_models = get_allegro_model(cfgs.allegro_model_path)

    end_points = {'sinput': sinput, 'point_clouds': [sinput.F]}
    with torch.no_grad():
        end_points = net(end_points)
        grasp_preds, grasp_features = parse_preds(end_points, use_v2=cfgs.use_graspnet_v2)
        # print('grasp preds:',grasp_preds[0])
    if len(grasp_preds) == 0:
        print("No grasp predicted.")
        return None

    # 4. 后处理
    grasp_preds = grasp_preds[0].cpu().numpy()
    grasp_features = grasp_features[0].cpu().numpy()
    grasp_preds, if_flip = flip_ggarray(grasp_preds)
    grasp_features = np.c_[grasp_features, if_flip]

    # 排序前 2000
    top_indices = grasp_preds[:, 0].argsort()[::-1][:2000]
    grasp_preds = grasp_preds[top_indices]
    grasp_features = grasp_features[top_indices]

    grasp_features_dic = get_graspgroup_features(grasp_features, [sinput])

    allegro_depths, allegro_types, scores, grasp_preds, grasp_features = get_allegro_depth_type(
        allegro_models, grasp_features_dic, grasp_preds, grasp_features
    )
    print('allegro grasp:',grasp_preds[0].shape)
    if not RANDOM:
        mask = scores > 0.7
        if not np.any(mask):
            print("❌ No high-score grasp pose.")
            return None
        grasp_preds = grasp_preds[mask]
        grasp_features = grasp_features[mask]
        allegro_depths = allegro_depths[mask]
        allegro_types = allegro_types[mask]
        scores = scores[mask]

    grasp_preds, if_flip = flip_z_ggarray(grasp_preds, allegro_types)

    # 构造抓取对象
    two_fingers_gg = GraspGroup(grasp_preds)
    allegro_gg = AllegroGraspGroup()
    allegro_gg.from_graspgroup(two_fingers_gg, allegro_types, cfgs.Allegro_mesh_json_path)
    allegro_gg.scores = scores
    allegro_gg.depths = allegro_gg.depths + allegro_depths + DEFAULT_DEPTH

    # 获取得分最高的抓取
    idx = np.argsort(scores)[::-1][0]
    grasp = allegro_gg[idx]

    pose = {
        "rotation_matrix": grasp.rotation_matrix,
        "translation": grasp.translation,
        "score": grasp.score,
        "grasp_type": grasp.grasp_type,
        "depth": grasp.depth,
        "angle": grasp.angle,
    }

    print("\n✅ Allegro Grasp Pose (From Raw PointCloud):")
    print(f"Score     : {pose['score']:.4f}")
    print(f"Grasp Type: {pose['grasp_type']}")
    print(f"Depth     : {pose['depth']:.4f} m")
    print(f"Angle     : {pose['angle']:.4f} rad")
    print(f"Translation (x,y,z): {pose['translation']}")
    print(f"Rotation Matrix:\n{pose['rotation_matrix']}")

    return pose`

**Comment by RogerRafa** [comment]
Sorry, I only tried using the DH3 to grasp.
